redhat-operators/codeready-workspaces/codeready-workspaces.1.1.0.csv.yaml

```diff
@@ -0,0 +1,345 @@
+      name: Eugene Ivantsov
+  description: >
+    A collaborative Kubernetes-native development solution that delivers OpenShift workspaces and in-browser IDE for rapid cloud application development.
+    This operator installs PostgreSQL, Red Hat SSO, and the CodeReady Workspaces server, as well as configures all three services.
+
+    ## Pre-Reqs
+
+    The operator service account requires cluster-admin privileges if you enable either `server.selfSignedCerts` or `auth.openShiftoAuth`.
+    After the operator is installed, grant the **codeready-operator** service account such privileges:
+
+
+    ```
+    oc adm policy add-cluster-role-to-user cluster-admin -z codeready-operator -n=$CodeReadyNamespace
+    ```
+
+    ## How to Install
+
+    Press the **Install** button, choose the upgrade strategy, and wait for the **Installed** Operator status.
+
+    When the operator is installed, create a new CR of Kind CheCluster (click the **Create New** button).
+    The CR spec contains all defaults (see below).
+
+    You can start using CodeReady Workspaces when the CR status is set to **Available**, and you see a URL to CodeReady Workspaces.
+
+    ## Defaults
+
+    By default, the operator deploys CodeReady Workspaces with:
+
+    * Bundled PostgreSQL and Red Hat SSO
+
+    * Common PVC strategy
+
+    * Auto-generated passwords
+
+    * HTTP mode (non-secure routes)
+
+    * Regular login (no login with OpenShift)
+
+    ## Installation Options
+
+    CodeReady Workspaces operator installation options include:
+
+    * Connection to external database and Keycloak/Red Hat SSO
+
+    * Configuration of default passwords and object names
+
+    * TLS mode
+
+    * PVC strategy (once shared PVC for all workspaces, PVC per workspace, or PVC per volume)
+
+    * Authentication options
+
+    ### External Database and Keycloak
+
+    To instruct the operator to skip deploying PostgreSQL and Red Hat SSO and connect to an existing DB and Red Hat SSO or Keycloak instead:
+    
+    * set respective fields to `true` in a custom resource spec
+    * provide the operator with connection and authentication details:
+
+
+
+    `externalDb: true`
+
+
+    `chePostgresHostname: 'yourPostgresHost'`
+
+
+    `chePostgresPort: '5432'`
+
+
+    `chePostgresUser: 'myuser'`
+
+
+    `chePostgresPassword: 'mypass'`
+
+
+    `chePostgresDb: 'mydb'`
+
+
+    `externalKeycloak: true`
+
+
+    `keycloakURL: 'https://my-keycloak.com'`
+
+
+    `keycloakRealm: 'myrealm'`
+
+
+    `keycloakClientId: 'myClient'`
+
+
+    ### TLS Mode
+
+    To activate TLS mode, set the respective field in the CR spec to `true` (in the `server` block):
+
+
+    ```
+    tlsSupport: true
+    ```
+
+    #### Self-signed Certificates
+
+    To use CodeReady Workspaces with TLS enabled, but the OpenShift router does not use certificates signed by a public authority, you can use self-signed certificates, which the operator can create for you (requires cluster-admin privileges):
```

Comment:
```suggestion
    If you want to use CodeReady Workspaces with TLS enabled and self-signed certificates, the operator can create those certs for you (this requires cluster-admin privileges):
```

Reply:
Let's try to avoid starting sentences with 'if'. We want the text to be authoritative and concise. "If you want" is redundant.

Reply:
If is pretty clear for developers - it's like everyone's favourite conditional block :)

I've tried to reword so that it's clear (original wasn't).

Reply:
```suggestion
    To use CodeReady Workspaces with TLS enabled and self-signed certificates (publicly signed certs are preferred), the operator can create them for you as long as you have cluster-admin privileges:
```